action/rlp_tx.go

```diff
@@ -24,11 +23,8 @@ func rlpSignedHash(tx *types.Transaction, signer types.Signer, sig []byte) (hash
 	if err != nil {
 		return hash.ZeroHash256, err
 	}
-	h := sha3.NewLegacyKeccak256()
-	if err = rlp.Encode(h, signedTx); err != nil {
-		return hash.ZeroHash256, err
-	}
-	return hash.BytesToHash256(h.Sum(nil)), nil
+	h := signedTx.Hash()
+	return hash.BytesToHash256(h[:]), nil
```

Comment:
`signedTx.Hash()` is a compatible change
for LegacyTx, its code is same as left side

Reply:
Is there a corresponding unit test to ensure this?

Reply:
in `TestRlpDecodeVerify`, hash of tx is calculated and verified match the testdata. 
If the code here is not compatible, `TestRlpDecodeVerify` would fail

Reply:
better to be separated in another pr